### ISSUE TITLE
Relative path specified dependencies are being installed as dev, without sub dev-depends

### ISSUE THREAD

**Issue by jdfreder** [npm/npm#10794]
Hi all,

I tried to find an open issue describing the problem I'm seeing and was unable to.  I have one main npm package and a few smaller npm packages that demonstrate how that package can be used.

The folder structure looks like this:

```
main/package.json
examples/a/package.json
examples/b/package.json
examples/c/package.json
```

Each of the example package.json files looks roughly like this:

```json
{
  "name": "example-a",
  "version": "1.0.0",
  "description": "...",
  "main": "index.js",
  "scripts": {
    "prepublish": "browserify"
  },
  "author": "...",
  "license": "BSD-3-Clause",
  "dependencies": {
    "main": "file:../../main",
  },
  "devDependencies": {
    "browserify": "^11.2.0"
  }
}
```

The main package looks something like this:

```json
{
  "name": "main",
  "version": "1.0.0",
  "description": "...",
  "main": "index.js",
  "scripts": {
    "prepublish": "babel"
  },
  "author": "...",
  "license": "BSD-3-Clause",
  "dependencies": {
  },
  "devDependencies": {
    "babel": "^6.3.13"
  }
}
```

One would think that you could run `npm install` inside one of the example project folders without hiccups.  However, doing so will yield an error, for this example, that `babel` doesn't exist.  It makes sense to me that `babel` is needed, because the relative path implies that you are installing from source, which means a dev install of the *main* package is needed.  However, what doesn't make sense is that either A) the dev dependencies of the *main* package aren't being installed or B) they are being installed, but to somewhere that is not on the path when the prepublish step of the *main* package is ran.

Thanks in advanced!

Also, for those who are curious, what I'm working on is open source.  So you can see the full failing example here: [github.com/ipython/ipywidgets](https://github.com/ipython/ipywidgets)

**Comment by jdfreder** [comment]
## TL;DR

I can provide an MVP repo if required...

**Comment by jdfreder** [comment]
I think I've identified the root issue in #10840 and (made the issue more concise).
